lib/iris/fileformats/netcdf/loader.py

```diff
@@ -587,7 +587,7 @@ def load_cubes(file_sources, callback=None, constraints=None):
                 warnings.warn("{}".format(e))
 
             # Perform any user registered callback function.
-            cube = run_callback(callback, cube, cf_var, cf.filename)
+            cube = run_callback(callback, cube, cf_var, file_source)
```

Comment:
I think this makes more sense.
I believe it is consistent with what we do for UM "structured loading"
see "Use of callbacks" in [structured_um_loading api docs](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris/fileformats/um.html?highlight=structured#iris.fileformats.um.structured_um_loading)